falcon_multiqc/commands/save.py

```diff
@@ -2,54 +2,91 @@
 import json
 import datetime
 from database.crud import session_scope
-from database.models import Base, RawData, Batch, Sample
+from database.models import Base, RawData, Batch, Sample, Cohort
 
 
 @click.command()
 @click.option("-d", "--directory", type=click.Path(), required=True, help="Path to multiqc_output directory")
-@click.option("-c", "--cohort", type=click.INT, required=True, help="Id of this batch's cohort (use create_cohort if you need a new one)")
-def cli(directory, cohort):
+@click.option("-s", "--sample_metadata", type=click.File(), required=True, help="Sample metadata file")
+@click.option("-b", "--batch_description", type=click.STRING, required=False, help="Description of this batch (assuming this is one batch)")
+@click.option("-c", "--cohort_description", type=click.STRING, required=False, help="Description of this cohort (assuming this is one cohort)")
+def cli(directory, sample_metadata, batch_description, cohort_description):
     """Saves the given directory to the falcon_multiqc database"""
-
+    # TODO: better error handling
     with open(directory + "/multiqc_data/multiqc_data.json") as multiqc_data:
-        multiqc_data_json = json.load(multiqc_data)
-
+        # Skip header
+        next(sample_metadata)
         with session_scope() as session:
-            batch_row = Batch(
-                cohort_id=cohort,
-                flow_cell_id=0,  # TODO: implement real flow cell id
-                path=directory,
-                date=datetime.datetime.now()  # TODO: implement real date
-            )
+            # Keep track of samples added, so we know its primary key, when saving raw data later.
+            samples = {}  # name : primary key id
+            cohort_id = None
 
-            session.add(batch_row)
-            session.flush()
+            for line in sample_metadata:
+                split = line.split(",")
+                sample_name = split[0]
+                batch_name = split[2]
+                flowcell_lane = split[3]
+                library_id = split[4]
+                platform = split[5]
+                centre = split[6]
+                reference = split[7]
+                type = split[8]
+                description = split[9]
 
-            # Keep track of samples added, so we don't duplicate any.
-            samples = {}  # name : primary key id
+                batch_id = None
+
+                if not cohort_id and session.query(Cohort.id).filter_by(id=split[1]).scalar() is None:
+                    # Cohort does not exist in database.
+                    cohort_row = Cohort(
+                        id=split[1],
+                        type=split[7],
+                        description=cohort_description
+                    )
+                    session.add(cohort_row)
+
+                # Get cohort id / name from the first data row (assuming the metadata is for 1 cohort).
+                if not cohort_id:
+                    cohort_id = split[1]
```

Comment:
Just a suggestion, 
![image](https://user-images.githubusercontent.com/71308285/96261398-21580280-100c-11eb-9860-3842e2887271.png)
And still have the same functionality, not sure which is better, up to you~

Reply:
Thanks I like that, done! There's also a suggestion button in Github btw if you wanna make suggestions.